projects/packages/post-list/src/class-post-list.php

```diff
@@ -80,43 +67,43 @@ public function enqueue_scripts( $hook ) {
 				plugin_dir_url( __DIR__ ) . 'build/style.rtl.css'
 			);
 
-			wp_set_script_translations( 'jetpack_posts_list_ui_script', 'jetpack' );
-
-			add_action( 'admin_footer', array( $this, 'print_post_data' ) );
+			add_filter( 'manage_posts_columns', array( $this, 'add_thumbnail_column' ) );
+			add_action( 'manage_posts_custom_column', array( $this, 'populate_thumbnail_rows' ), 10, 2 );
 		}
 	}
 
 	/**
-	 * Outputs a JSON blob to the global `wp_admin_posts` variable, for use
-	 * by the JS application
+	 * Adds a new column header for displaying the thumbnail of a post.
+	 *
+	 * @param array $columns An array of column names.
+	 * @return array An array of column names.
 	 */
-	public function print_post_data() {
-		global $wp_query;
+	public function add_thumbnail_column( $columns ) {
+		$new_column = array( 'thumbnail' => '<span>' . __( 'Thumbnail', 'jetpack' ) . '</span>' );
+		$keys       = array_keys( $columns );
+		$index      = array_search( 'title', $keys, true );
+		$pos        = false === $index ? count( $columns ) : $index;
```

Comment:
Might we want to just return `$columns` here?

Reply:
🤔 I'm not sure I follow. This function inserts $new_column into $columns and returns the resulting array. I believe $columns is the original input array unmodified. Do I misunderstand?

Reply:
Yep. If the title field is not found we set it so the thumbnail is inserted at the end. That seemed a bit odd to me and I wondered if we'd just want to not insert the thumbnail column at all.

Reply:
Ok. I'm following you now. Yeah, that probably makes more sense. Title shouldn't be missing, but if it is missing something is wrong, and we probably shouldn't try adding thumbnails. I changed it and added a new unit test for this case.